components/Header.js

```diff
@@ -32,11 +37,17 @@ const StyledHeader = styled.h1`
   justify-content: start;
   align-items: center;
   font-size: 1rem;
-  color: #4686cd;
+  color: var(--text-color-dark);
   height: 44px;
   margin: 0;
   padding-inline: 16px;
   background-color: ${(props) =>
     props.isScrolled ? "rgba(245, 245, 245, 0.8)" : "rgba(255, 255, 255, 0.5)"};
```

Comment:
**Here is a bug 🐛**  
Update `props.isScrolled` to `props.$isScrolled` in the `StyledHeader` component to match the prop name:

 💡 **Note:** This issue may not have been immediately apparent because the background color change is subtle.
It’s helpful to use Dev Tools to inspect style rules on dynamic elements to confirm they update as expected. 

```suggestion
  background-color: ${(props) =>
    props.$isScrolled ? "rgba(245, 245, 245, 0.8)" : "rgba(255, 255, 255, 0.5)"};
```